enclave-tls/src/tls_wrappers/openssl/un_negotiate.c

```diff
@@ -229,9 +232,17 @@ int verify_certificate(void *ctx, uint8_t *der_cert, uint32_t der_cert_len)
 #else
 int verify_certificate(int preverify, X509_STORE_CTX *ctx)
 {
+	int *etls_ex_data;
 	X509_STORE *cert_store = X509_STORE_CTX_get0_store(ctx);
-	tls_wrapper_ctx_t *tls_ctx = X509_STORE_get_ex_data(cert_store, 0);
-	ETLS_DEBUG("verify_certification's tls_ctx is %x\n", tls_ctx);
+	etls_ex_data = pthread_getspecific(g_key);
+	if (!etls_ex_data) {
+		ETLS_DEBUG("etls_ex_data is NULL pointer\n");
+		etls_ex_data = malloc(sizeof(int));
+		if (!etls_ex_data)
+			ETLS_ERR("failed to malloc etls_ex_data\n");
+	}
```

Comment:
If pthread_getspecific() returns NULL, it means openssl_internal_negotiate() is not called correctly and a serious problem happened, so why not just emit a fatal ?

Reply:
will return 0